src/Exception/BatchException.php

```diff
@@ -2,61 +2,136 @@
 
 namespace Http\Client\Exception;
 
+use Http\Client\BatchResult;
+use Http\Client\Exception;
+use Psr\Http\Message\RequestInterface;
+
 /**
  * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
  */
-final class BatchException extends TransferException
+final class BatchException extends RuntimeException
 {
     /**
-     * @var TransferException[]
+     * @var BatchResult
+     */
+    private $result;
+
+    /**
+     * @var \SplObjectStorage
      */
     private $exceptions;
 
+    public function __construct()
+    {
+        $this->exceptions = new \SplObjectStorage();
+    }
+
     /**
-     * @param TransferException[] $exceptions
+     * Returns a list BatchResult which contains succesful responses
```

Comment:
Returns the BatchResult that contains those responses that where successful.

Note that the BatchResult may contains 0 responses if all requests failed.